src/rdbms/mongoose_rdbms_odbc.erl

```diff
@@ -191,20 +191,38 @@ tabcol_to_odbc_type(TabCol = {Table, Column}, TableDesc) ->
             ODBCType
     end.
 
+%% Null should be encoded with the a correct type.
+%% Otherwice inserting two records, where one value is null and another is
+%% not null, would case:
+%% [FreeTDS][SQL Server]Conversion failed when converting the nvarchar value
+%% 'orig_id' to data type int. SQLSTATE IS: 22018
+unicode_mapper(null) ->
```

Comment:
Fixes 
https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3466/96375/odbc_mssql_mnesia.24.0.5/big/ct_run.test%407759d557a382.2021-12-21_10.09.04/big_tests.tests.mam_SUITE.logs/run.2021-12-21_10.12.19/mam_suite.retract_wrong_message.18053.html

Probably ODBC driver is "very" smart: if we insert several records, it reuses types.